homeassistant/components/weather/__init__.py

```diff
@@ -179,6 +179,10 @@ def state_attributes(self):
             forecast = []
             for forecast_entry in self.forecast:
                 forecast_entry = dict(forecast_entry)
+                if isinstance(forecast_entry[ATTR_FORECAST_TIME], datetime):
```

Comment:
Each integration platform is responsible for returning the correct type and format.

Reply:
from my rough analysis 50% return string 50% return a datetime object.

Would a PR to change those 50% returning a datetime to isoformat string be best ? can I make it 1 PR or should it be split across integrations ?

Reply:
Usually better to split into multiple PRs. Depends a bit on how large the change is.

Reply:
just this cast (1 line?), but across all the weather entities

Reply:
Probably ok in one PR.